lib/chewy/type.rb

```diff
@@ -41,6 +41,18 @@ def self.type_name
       adapter.type_name
     end
 
+    # Returns the identifier value
+    #
+    def self.identify objects
+      if root_object.id
+        identifiers_from_evaluator = root_object.compose_id(objects)
+      else
+        identifiers_from_evaluator = nil
```

Comment:
The thing here is that `identifiers_from_evaluator` will be nil without explicit assignment.

`identifiers_from_evaluator = root_object.compose_id(objects) if root_object.id` would be enough